src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2502 @@
+            [`AutoFeatureExtractor.__call__`] for details.
+
+        pixel_mask (`torch.LongTensor` of shape `(batch_size, height, width)`, *optional*):
+            Mask to avoid performing attention on padding pixel values. Mask values selected in `[0, 1]`:
+
+            - 1 for pixels that are real (i.e. **not masked**),
+            - 0 for pixels that are padding (i.e. **masked**).
+
+            [What are attention masks?](../glossary#attention-mask)
+
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of Detr's decoder attention layers.
+
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~MaskFormerModelOutput`] instead of a plain tuple.
+"""
+
+
+class MaskFormerPretrainedModel(PreTrainedModel):
+    config_class = MaskFormerConfig
+    base_model_prefix = "model"
+    main_input_name = "pixel_values"
+
+    def _init_weights(self, module: nn.Module):
+        xavier_std = self.config.init_xavier_std
+        std = self.config.init_std
+        if isinstance(module, MaskFormerTransformerModule):
+            if module.input_projection is not None:
+                nn.init.xavier_uniform_(module.input_projection.weight, gain=xavier_std)
+                nn.init.constant_(module.input_projection.bias, 0)
+        # FPN
+        elif isinstance(module, MaskFormerFPNModel):
+            nn.init.xavier_uniform_(module.stem[0].weight, gain=xavier_std)
+
+        elif isinstance(module, MaskFormerFPNLayer):
+            nn.init.xavier_uniform_(module.proj[0].weight, gain=xavier_std)
+
+        elif isinstance(module, MaskFormerFPNConvLayer):
+            nn.init.xavier_uniform_(module[0].weight, gain=xavier_std)
+        # The MLP head
+        elif isinstance(module, MaskformerMLPPredictionHead):
+            # I was not able to find the correct initializer in the original implementation
+            # we'll use xavier
+            for layer in module:
+                nn.init.xavier_uniform_(layer[0].weight, gain=xavier_std)
+                nn.init.constant_(layer[0].bias, 0)
+        # copied from swin
+        if isinstance(module, nn.Linear):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.backbone_config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        # copied from DETR
+        if isinstance(module, (nn.Linear, nn.Conv2d, nn.BatchNorm2d)):
```

Comment:
Linear will already have been taken from `swin`, but I don't think it's a big deal.
```suggestion
        if isinstance(module, (nn.Conv2d, nn.BatchNorm2d)):
```

Reply:
Due to the control flow, `nn.Linear` needs to be there

Reply:
It will erase the part from swin then.

Reply:
In this case, what do you think is the best way to initialize only the correct nested parts? I could add a check on the instance type e.g.

```python
if "Swin" in module.__name__:
 .... 
 
 ```

Reply:
If that works, it's simple enough we can add it, good idea!

Reply:
Unfortunately, that introduces a lot of control flow to make sure the module goes in the correct branch:

```python
        if "Swin" in module.__class__.__name__:
            for module in module.modules():
                self._init_weights_swin(module)
        elif "Detr"  in module.__class__.__name__: 
            for module in module.modules():
                self._init_weights_detr(module, std=std)
        elif "MaskFormer" in module_name: 
            for module in module.modules():
                self._init_weights_maskformer(module)
```
I suggest we keep it the way it is

Reply:
Then you should remove the test above to avoid initializing the swin linear layers twice.

Reply:
Done